components/suggestion/switch_channel_provider.jsx

```diff
@@ -620,13 +621,19 @@ export default class SwitchChannelProvider extends Provider {
     fetchAndFormatRecentlyViewedChannels(resultsCallback) {
         const state = getState();
         const recentChannels = getChannelsInAllTeams(state).concat(getDirectAndGroupChannels(state));
-        const channels = this.wrapChannels(recentChannels, Constants.MENTION_RECENT_CHANNELS);
-        if (channels.length === 0) {
+        const wrappedRecentChannels = this.wrapChannels(recentChannels, Constants.MENTION_RECENT_CHANNELS);
+        const lastUnreadChannel = state.views.channel.lastUnreadChannel;
+        const unreadChannels = getSortedAllTeamsUnreadChannels(state, lastUnreadChannel).slice(0, 5);
```

Comment:
I don't think we need to pass the `lastUnreadChannel` in here. It's only needed in the LHS to ensure that, when you switch to an unread channel, it stays in the same place in the LHS until you switch away from it. The channel switcher isn't visible all the time, so I don't think it needs that.